src/lib/storage.ts

```diff
@@ -121,6 +127,31 @@ export class Storage implements PeprStore {
     });
   };
 
+  /**
+   * Creates a promise and subscribes to the store, the promise resolves when
+   * the key is removed from the store.
+   *
+   * @param key - The key to add into the store
+   * @returns
+   */
+  removeItemAndWait = (key: string) => {
+    this.#dispatchUpdate("remove", [key]);
+    return new Promise<void>((resolve, reject) => {
+      const unsubscribe = this.subscribe(data => {
+        if (!data.hasOwnProperty(key)) {
```

Comment:
```suggestion
        if (!Object.hasOwn(data, key)) {
```

Maybe ESLint will like this better..?
- https://stackoverflow.com/a/72895649
- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwn